app/players/dwarves/dwarf.py

```diff
@@ -0,0 +1,12 @@
+from abc import ABC, abstractmethod
+from app.players.player import Player
+
+
+class Dwarf(Player, ABC):
+    def __init__(self, nickname: str, favourite_dish: str) -> None:
+        super().__init__(nickname)
+        self._favourite_dish = favourite_dish
+
+    @abstractmethod
```

Comment:
The Dwarf class doesn't need to be abstract because it doesn't define any additional abstract methods itself. Instead, it simply implements the eat_favourite_dish method, which is required by its subclasses (DwarfWarrior and DwarfBlacksmith). Since the task specifies that Dwarf will be a concrete class that provides common functionality for its subclasses, we don't need to make Dwarf an abstract base class (ABC). But it also works with ABC as you wrote.